persistence-migration/src/main/resources/migration/changelog-v5.6.0.xml

```diff
@@ -2353,4 +2353,262 @@
+                  INTO project_ids
+                  FROM (
+                    SELECT upr."PROJECT_ID"
+                      FROM "USER_PROJECT_ROLES" upr
+                     INNER JOIN new_table
+                        ON new_table."ROLE_ID" = upr."ROLE_ID"
+                      FULL OUTER JOIN old_table
+                        ON new_table."ROLE_ID" = old_table."ROLE_ID"
+                  ) sub;
+              ELSE
+                SELECT ARRAY_AGG(DISTINCT "PROJECT_ID")
+                  INTO project_ids
+                  FROM "USER_PROJECT_ROLES"
+                 WHERE "ROLE_ID" = ANY(role_ids);
+              END IF;
+
+              PERFORM recalc_user_project_effective_permissions(project_ids);
+              RETURN NULL;
+            END;
+            $$ LANGUAGE plpgsql;
+        </sql>
+
+        <sql splitStatements="true">
+            -- INSERT trigger for USER_PROJECT_ROLES
+            CREATE TRIGGER trigger_effective_permissions_mx_on_user_project_roles_insert
+            AFTER INSERT ON "USER_PROJECT_ROLES"
+            REFERENCING NEW TABLE AS new_table
+            FOR EACH STATEMENT
+            EXECUTE FUNCTION role_effective_permissions_mx_on_insert();
+
+            -- DELETE trigger for USER_PROJECT_ROLES
+            CREATE TRIGGER trigger_effective_permissions_mx_on_user_project_roles_delete
+            AFTER DELETE ON "USER_PROJECT_ROLES"
+            REFERENCING OLD TABLE AS old_table
+            FOR EACH STATEMENT
+            EXECUTE FUNCTION role_effective_permissions_mx_on_delete();
+
+            -- UPDATE trigger for USER_PROJECT_ROLES
+            CREATE TRIGGER trigger_effective_permissions_mx_on_user_project_roles_update
+            AFTER UPDATE ON "USER_PROJECT_ROLES"
+            REFERENCING OLD TABLE AS old_table NEW TABLE AS new_table
+            FOR EACH STATEMENT
+            EXECUTE FUNCTION role_effective_permissions_mx_on_update();
+
+            -- INSERT trigger for ROLES_PERMISSIONS
+            CREATE TRIGGER trigger_effective_permissions_mx_on_roles_permissions_insert
+            AFTER INSERT ON "ROLES_PERMISSIONS"
+            REFERENCING NEW TABLE AS new_table
+            FOR EACH STATEMENT
+            EXECUTE FUNCTION role_effective_permissions_mx_on_insert();
+
+            -- DELETE trigger for ROLES_PERMISSIONS
+            CREATE TRIGGER trigger_effective_permissions_mx_on_roles_permissions_delete
+            AFTER DELETE ON "ROLES_PERMISSIONS"
+            REFERENCING OLD TABLE AS old_table
+            FOR EACH STATEMENT
+            EXECUTE FUNCTION role_effective_permissions_mx_on_delete();
+
+            -- UPDATE trigger for ROLES_PERMISSIONS
+            CREATE TRIGGER trigger_effective_permissions_mx_on_roles_permissions_update
+            AFTER UPDATE ON "ROLES_PERMISSIONS"
+            REFERENCING OLD TABLE AS old_table NEW TABLE AS new_table
+            FOR EACH STATEMENT
+            EXECUTE FUNCTION effective_permissions_mx_on_update();
```

Comment:
Argh, only spotted this last-minute. I think this should call a different function?

```suggestion
            EXECUTE FUNCTION role_effective_permissions_mx_on_update();
```

The `effective_permissions_mx_on_update` only inspects the `PROJECT_ACCESS_TEAMS` table and wouldn't "see" the change that fired this trigger.